src/Xamarin.Android.Tools.ApiXmlAdjuster/JavaApiXmlLoaderExtensions.cs

```diff
@@ -67,11 +70,11 @@ public static void Load (this JavaPackage package, XmlReader reader, bool isRefe
 					if (reader.LocalName == "class") {
 						var kls = new JavaClass (package) { IsReferenceOnly = isReferenceOnly };
 						kls.Load (reader);
-						package.Types.Add (kls);
```

Comment:
…back on the "smaller diffs!" front, if we *did* have a `class JavaTypes : KeyedCollection<string, JavaType>` class, this statement wouldn't need to change, nor the many other statements just like it…

Reply:
I tested out `KeyedCollection`, but it cannot handle duplicate keys:
```
System.ArgumentException: 'An item with the same key has already been added. Key: key1'
```